apps/survey/src/util/meal-food.ts

```diff
@@ -191,3 +194,55 @@ export function flattenFoods(foods: FoodState[]): FoodState[] {
 
   return result;
 }
+
+/**
+ * Check if a prompt's custom conditions are met for the given food
+ */
+function checkPromptCustomConditions(store: SurveyStore, mealState: MealState, foodState: FoodState, prompt: Prompt): boolean {
```

Comment:
I think these checks could rather go to `prompt-filters`. I have a feeling that we're creating to much circulars deps dragging `evaluateCondition` from promptMgr to simple util functions.

...ultimate I think we need to clean up prompt manager is too fat and extract `evaluateCondition` etc out of there.